tests/qiskit_convert_test.py

```diff
@@ -1193,6 +1193,80 @@ def test_nonregister_bits() -> None:
         tk_to_qiskit(c)
 
 
+# https://github.com/CQCL/pytket-qiskit/issues/415
+def test_ifelseop_two_branches() -> None:
+    qreg = QuantumRegister(2, "r")
+    creg = ClassicalRegister(2, "s")
+    circuit = QuantumCircuit(qreg, creg)
+    (q0, q1) = qreg
+    (c0, c1) = creg
+
+    circuit.h(q0)
+    circuit.measure(q0, c0)
+
+    with circuit.if_test((c0, 1)) as else_:
+        circuit.h(q1)
+    with else_:
+        circuit.x(q1)
+    circuit.measure(q1, c1)
+
+    tkc = qiskit_to_tk(circuit)
+    assert tkc.n_gates_of_type(OpType.Conditional) == 2
+    if_cond, else_cond = tuple(tkc.commands_of_type(OpType.Conditional))
+
+    if_circ = if_cond.op.op.get_circuit()
+    else_circ = else_cond.op.op.get_circuit()
+
+    exp_if_circ = Circuit()
+    r = exp_if_circ.add_q_register("r", 1)
+    exp_if_circ.H(r[0])
+
+    exp_else_circ = Circuit()
+    s = exp_else_circ.add_q_register("s", 1)
+    exp_else_circ.H(s[0])
+
+    assert if_circ == exp_if_circ
```

Comment:
something weird is going on with register names here so this. equality check fails.

`if_circ`
![if1](https://github.com/user-attachments/assets/f909c076-3d76-4750-9ed8-d41829906999)

`exp_if_circ`
![if2](https://github.com/user-attachments/assets/c612d3c6-82ee-45c4-9473-6091737821a4)

Reply:
Ah looking more closely at the original circuit this is actually expected as the Hadamard is applied to qubit one.

Creating a fresh pytket circuit the register starts from zero so thats why the comparison fails (expected)

I've added a simpler gate counting test for now.